src/Marten/Internal/CompiledQueries/NewCode.cs

```diff
@@ -0,0 +1,437 @@
+    {
+        QueryType = queryType;
+        OutputType = outputType;
+
+        sortMembers();
+    }
+
+    #region finding members on query type
+
+    private void sortMembers()
+    {
+        foreach (var member in findMembers())
+        {
+            var memberType = member.GetRawMemberType();
+            if (memberType == typeof(QueryStatistics))
+            {
+                StatisticsMember = member;
+            }
+
+            else if (memberType.Closes(typeof(IDictionary<,>)))
+            {
+                IncludeMembers.Add(member);
+            }
+            else if (memberType.Closes(typeof(Action<>)))
+            {
+                IncludeMembers.Add(member);
+            }
+            else if (memberType.Closes(typeof(IList<>)))
+            {
+                IncludeMembers.Add(member);
+            }
+            else if (memberType.IsNullable())
+            {
+                InvalidMembers.Add(member);
+            }
+            else if (QueryCompiler.Finders.All(x => !x.Matches(memberType)))
+            {
+                InvalidMembers.Add(member);
+            }
+            else if (member is PropertyInfo)
+            {
+                var queryMember = typeof(PropertyQueryMember<>).CloseAndBuildAs<IQueryMember>(member, memberType);
+                QueryMembers.Add(queryMember);
+            }
+            else if (member is FieldInfo)
+            {
+                var queryMember = typeof(FieldQueryMember<>).CloseAndBuildAs<IQueryMember>(member, memberType);
+                QueryMembers.Add(queryMember);
+            }
+        }
+    }
+
+    private IEnumerable<MemberInfo> findMembers()
+    {
+        foreach (var field in QueryType.GetFields(BindingFlags.Instance | BindingFlags.Public)
+                     .Where(x => !x.HasAttribute<MartenIgnoreAttribute>())) yield return field;
+
+        foreach (var property in QueryType.GetProperties(BindingFlags.Instance | BindingFlags.Public)
+                     .Where(x => !x.HasAttribute<MartenIgnoreAttribute>())) yield return property;
+    }
+
+    #endregion
+
+    public string TenantId { get; set; }
+
+    #region ICommandBuilder implementation
+
+    private CommandPlan _current;
+
+    string ICommandBuilder.LastParameterName => _current.Parameters.LastOrDefault()?.Parameter.ParameterName;
+
+    private CommandPlan appendCommand()
+    {
+        var plan = new CommandPlan();
+        _commands.Add(plan);
+
+        return plan;
+    }
+
+    void ICommandBuilder.Append(string sql)
+    {
+        _current ??= appendCommand();
+
+        _current.CommandText += sql;
+    }
+
+    void ICommandBuilder.Append(char character)
+    {
+        _current ??= appendCommand();
+        _current.CommandText += character;
+    }
+
+    private int _parameterIndex = 0;
+
+    NpgsqlParameter ICommandBuilder.AppendParameter(object value)
+    {
+        _current ??= appendCommand();
+        var name = "p" + _parameterIndex;
```

Comment:
The native parameter style of Postgres is `$i` (where `i` denotes the index), so it's also recommended by [Npgsql docs](https://www.npgsql.org/doc/basic-usage.html#parameters).

This produces a named parameter `p1`.
Can you change this to being `$1` in order to avoid re-parsing the SQL in the Npgsql-layer?

Reply:
That's never executed. That's strictly to be able to "plan" the compiled query command.